src/System.Management.Automation/engine/CommandProcessorBase.cs

```diff
@@ -661,6 +664,58 @@ internal void DoComplete()
             }
         }
 
+        protected virtual void HandleScopedAction(Action action, string traceMessage)
```

Comment:
Consider:

```suggestion
        protected virtual void HandleScopedAction<T>(Action<T> action, T state, string traceMessage)
```

So that you can avoid closures later:

```csharp
HandleScopedAction(cmdlet => cmdlet.Dispose(), scriptCmdlet, "something");
```